packages/flutter_tools/lib/src/commands/test.dart

```diff
@@ -508,6 +520,26 @@ class TestCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts {
     return FlutterCommandResult.success();
   }
 
+  Set<String> _getCoveragePackages(
+    List<String> packagesRegExps,
+    FlutterProject flutterProject,
+    PackageConfig packageConfig,
+  ) {
+    final String projectName = flutterProject.manifest.appName;
+    final Set<String> packagesToInclude = <String>{
+      if (packagesRegExps.isEmpty) projectName,
+    };
+    for (final String regExpStr in packagesRegExps) {
+      final RegExp regExp = RegExp(regExpStr);
+      packagesToInclude.addAll(
+        packageConfig.packages
+            .map((Package e) => e.name)
+            .where((String e) => regExp.hasMatch(e)),
+      );
+    }
```

Comment:
`RegExp` constructor can throw `FormatException`. Should we catch it and rethrow as `ToolExit`?

```suggestion
    try {
      for (final String regExpStr in packagesRegExps) {
        final RegExp regExp = RegExp(regExpStr);
        packagesToInclude.addAll(
          packageConfig.packages
              .map((Package e) => e.name)
              .where((String e) => regExp.hasMatch(e)),
        );
      }
    } on FormatException catch (e) {
      throwToolExit('Regular expression syntax is invalid. $e');
    }
```

Reply:
This SGTM, as long as the exception message itself shows the input pattern that was invalid.